client/extensions/woocommerce/state/sites/customers/selectors.js

```diff
@@ -1,86 +1,58 @@
+function inFlight( state, siteId, searchTerm ) {
+	return state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers.isSearching[ searchTerm ];
+}
+
 /**
  * @param {object} state Whole Redux state tree
  * @param {string} searchTerm Search term to check
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {boolean} Whether the search results for a given term have been successfully loaded from the server.
  */
-export const isCustomerSearchLoaded = (
-	state,
-	searchTerm,
-	siteId = getSelectedSiteId( state )
-) => {
-	const inFlight = get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.isSearching[${ searchTerm }]`
-	);
+export function isCustomerSearchLoaded( state, searchTerm, siteId = getSelectedSiteId( state ) ) {
 	// Strict check because it could also be undefined.
-	return false === inFlight;
-};
+	return inFlight( state, siteId, searchTerm ) === false;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {string} searchTerm Search term to check
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {boolean} Whether the search results for a given term are currently being retrieved from the server.
  */
-export const isCustomerSearchLoading = (
-	state,
-	searchTerm,
-	siteId = getSelectedSiteId( state )
-) => {
-	const inFlight = get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.isSearching[${ searchTerm }]`
-	);
+export function isCustomerSearchLoading( state, searchTerm, siteId = getSelectedSiteId( state ) ) {
 	// Strict check because it could also be undefined.
-	return true === inFlight;
-};
+	return inFlight( state, siteId, searchTerm ) === true;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {string} searchTerm Search term to check
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {Array} List of customer results matching term
  */
-export const getCustomerSearchResults = (
-	state,
-	searchTerm,
-	siteId = getSelectedSiteId( state )
-) => {
+export function getCustomerSearchResults( state, searchTerm, siteId = getSelectedSiteId( state ) ) {
 	if ( ! isCustomerSearchLoaded( state, searchTerm, siteId ) ) {
 		return [];
 	}
 
-	const customers = get( state, `extensions.woocommerce.sites[${ siteId }].customers.items`, {} );
-	const customerIdsForTerm = get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.queries[${ searchTerm }]`,
-		[]
-	);
-	return filter( customerIdsForTerm.map( id => customers[ id ] || false ) );
-};
+	const customers = state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers.items ?? {};
+	const customerIdsForTerm =
+		state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers.queries[ searchTerm ] ?? [];
+	return customerIdsForTerm.map( id => customers[ id ] || false ).filter( customer => !! customer );
```

Comment:
👍 It's nice that we're getting away from lodash's `filter` 👍

Reply:
It's a nice little win 🙂